tests/prediction_pipeline/test_prediction_pipeline_tiled.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import xarray as xr
+from numpy.testing import assert_array_almost_equal
+
+from bioimageio.core import load_resource_description
+from bioimageio.core.resource_io.nodes import Model
+from bioimageio.core.prediction import predict_with_tiling
+
+
+def _test_prediction_pipeline(model_package, weight_format):
+    from bioimageio.core.prediction_pipeline import create_prediction_pipeline
+
+    bio_model = load_resource_description(model_package)
+    assert isinstance(bio_model, Model)
+    pp = create_prediction_pipeline(bioimageio_model=bio_model, weight_format=weight_format)
+
+    inputs = [
+        xr.DataArray(np.load(str(test_tensor)), dims=tuple(spec.axes))
+        for test_tensor, spec in zip(bio_model.test_inputs, bio_model.inputs)
+    ]
+    outputs = list(predict_with_tiling(pp, inputs, True, verbose=False))
+    assert isinstance(outputs, list)
+
+    expected_outputs = [
+        xr.DataArray(np.load(str(test_tensor)), dims=tuple(spec.axes))
+        for test_tensor, spec in zip(bio_model.test_outputs, bio_model.outputs)
+    ]
+    assert len(outputs) == len(expected_outputs)
+
+    for out, exp in zip(outputs, expected_outputs):
+        assert_array_almost_equal(out, exp, decimal=4)
+
+
+def test_prediction_pipeline_torch(any_torch_model):
```

Comment:
I think we can't run these tests with `any...` model yet, because we have some models with multiple outputs etc. that are not supported by the tiled prediction yet.
Best to refactor the tests linked to above to here and extend them by the shape change model.

Reply:
Refactored my tests into `test_prediction.py` and made the tests specific with respect to the model. I added 2 models to the tiling tests that both pass, including my recently added simple test model.

On that note, this pull request breaks the existing `test_predict_image_with_tiling_fixed_output_shape` test since we now are more strict about throwing an error when trying to tile with fixed output shape.

Reply:
Thanks for the changes.

> On that note, this pull request breaks the existing `test_predict_image_with_tiling_fixed_output_shape` test since we now are more strict about throwing an error when trying to tile with fixed output shape.

Ok, I will think about that case tomorrow (a bit late here for this right now ;)).

Reply:
I think we need to enable the case of a explicit output shape if the input shape is also explicit and if they are identical. I made a small PR with these changes, see below.